superset-frontend/src/components/Select/Select.tsx

```diff
@@ -44,7 +43,21 @@ import { hasOption } from './utils';
 
 const { Option } = AntdSelect;
 
-type AntdSelectAllProps = AntdSelectProps<AntdSelectValue>;
+export type RawValue = string | number;
+
+export interface LabeledValue {
+  key?: string;
+  value: RawValue;
+  label: ReactNode;
+  customLabel?: ReactNode;
+}
+
+export type RawValues = RawValue[];
+export type LabeledValues = LabeledValue[];
+export type SelectValue = RawValue | LabeledValue | RawValues | LabeledValues;
```

Comment:
Just a nit, but do you think we could make `SelectValue` internal instead of exporting it? I can see the use cases for the other types but I think when we are using a Select we already defined if their values will be `RawValue` or `LabeledValue`. Considering that we are not mixing the types for the same Select.

Reply:
I'm fine with either way. But ideally the `Select` component itself should probably be generic. Similar to this: https://github.com/ktmud/chakra-ui-select/blob/main/Select/Select.tsx#L75-L78

Reply:
I like the use of generics as you mentioned. Maybe it can be a future improvement.